mask rcnn  dynamic batch inference pytorch model convert tensorrt  error

Thanks for your bug report. We appreciate it a lot.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://github.com/open-mmlab/mmdeploy/blob/master/docs/en/faq.md) but cannot get the expected help.
3. The bug has not been fixed in the latest version.

**Describe the bug**

python ./tools/deploy.py configs/mmdet/instance-seg/instance-seg_tensorrt-fp16_dynamic-320x320-1344x1344.py   E:/MWDeepLearning/3.AI_SoftWare/DeepLearning_Deploy/DeepLearning_Train/mmdetection/configs/mask_rcnn/mask_rcnn_r50_fpn_mstrain-poly_3x_coco.py  ./checkpoints/mask_rcnn_r50_fpn_mstrain-poly_3x_coco.pth  1.jpg   --work-dir work_dir --show  --device cuda:0
loading mmdeploy_execution ...
failed to load library mmdeploy_execution
loading mmdeploy_cpu_device ...
failed to load library mmdeploy_cpu_device
loading mmdeploy_cuda_device ...
failed to load library mmdeploy_cuda_device
loading mmdeploy_graph ...
failed to load library mmdeploy_graph
loading mmdeploy_directory_model ...
failed to load library mmdeploy_directory_model
loading mmdeploy_transform ...
failed to load library mmdeploy_transform
loading mmdeploy_cpu_transform_impl ...
failed to load library mmdeploy_cpu_transform_impl
loading mmdeploy_cuda_transform_impl ...
failed to load library mmdeploy_cuda_transform_impl
loading mmdeploy_transform_module ...
failed to load library mmdeploy_transform_module
loading mmdeploy_trt_net ...
failed to load library mmdeploy_trt_net
loading mmdeploy_net_module ...
failed to load library mmdeploy_net_module
loading mmdeploy_mmcls ...
failed to load library mmdeploy_mmcls
loading mmdeploy_mmdet ...
failed to load library mmdeploy_mmdet
loading mmdeploy_mmseg ...
failed to load library mmdeploy_mmseg
loading mmdeploy_mmocr ...
failed to load library mmdeploy_mmocr
loading mmdeploy_mmedit ...
failed to load library mmdeploy_mmedit
loading mmdeploy_mmpose ...
failed to load library mmdeploy_mmpose
loading mmdeploy_mmrotate ...
failed to load library mmdeploy_mmrotate
loading mmdeploy_execution ...
failed to load library mmdeploy_execution
loading mmdeploy_cpu_device ...
failed to load library mmdeploy_cpu_device
loading mmdeploy_cuda_device ...
failed to load library mmdeploy_cuda_device
loading mmdeploy_graph ...
failed to load library mmdeploy_graph
loading mmdeploy_directory_model ...
failed to load library mmdeploy_directory_model
loading mmdeploy_transform ...
failed to load library mmdeploy_transform
loading mmdeploy_cpu_transform_impl ...
failed to load library mmdeploy_cpu_transform_impl
loading mmdeploy_cuda_transform_impl ...
failed to load library mmdeploy_cuda_transform_impl
loading mmdeploy_transform_module ...
failed to load library mmdeploy_transform_module
loading mmdeploy_trt_net ...
failed to load library mmdeploy_trt_net
loading mmdeploy_net_module ...
failed to load library mmdeploy_net_module
loading mmdeploy_mmcls ...
failed to load library mmdeploy_mmcls
loading mmdeploy_mmdet ...
failed to load library mmdeploy_mmdet
loading mmdeploy_mmseg ...
failed to load library mmdeploy_mmseg
loading mmdeploy_mmocr ...
failed to load library mmdeploy_mmocr
loading mmdeploy_mmedit ...
failed to load library mmdeploy_mmedit
loading mmdeploy_mmpose ...
failed to load library mmdeploy_mmpose
loading mmdeploy_mmrotate ...
failed to load library mmdeploy_mmrotate
loading mmdeploy_execution ...
failed to load library mmdeploy_execution
loading mmdeploy_cpu_device ...
failed to load library mmdeploy_cpu_device
loading mmdeploy_cuda_device ...
failed to load library mmdeploy_cuda_device
loading mmdeploy_graph ...
failed to load library mmdeploy_graph
loading mmdeploy_directory_model ...
failed to load library mmdeploy_directory_model
loading mmdeploy_transform ...
failed to load library mmdeploy_transform
loading mmdeploy_cpu_transform_impl ...
failed to load library mmdeploy_cpu_transform_impl
loading mmdeploy_cuda_transform_impl ...
failed to load library mmdeploy_cuda_transform_impl
loading mmdeploy_transform_module ...
failed to load library mmdeploy_transform_module
loading mmdeploy_trt_net ...
failed to load library mmdeploy_trt_net
loading mmdeploy_net_module ...
failed to load library mmdeploy_net_module
loading mmdeploy_mmcls ...
failed to load library mmdeploy_mmcls
loading mmdeploy_mmdet ...
failed to load library mmdeploy_mmdet
loading mmdeploy_mmseg ...
failed to load library mmdeploy_mmseg
loading mmdeploy_mmocr ...
failed to load library mmdeploy_mmocr
loading mmdeploy_mmedit ...
failed to load library mmdeploy_mmedit
loading mmdeploy_mmpose ...
failed to load library mmdeploy_mmpose
loading mmdeploy_mmrotate ...
failed to load library mmdeploy_mmrotate
2022-09-05 10:57:13,126 - mmdeploy - INFO - Start pipeline mmdeploy.apis.pytorch2onnx.torch2onnx in subprocess
load checkpoint from local path: ./checkpoints/mask_rcnn_r50_fpn_mstrain-poly_3x_coco.pth
e:\mwdeeplearning\3.ai_software\deeplearning_deploy\deeplearning_train\mmdetection\mmdet\datasets\utils.py:70: UserWarning: "ImageToTensor" pipeline is replaced by "DefaultFormatBundle" for batch inference. It is recommended to manually replace it in the test data pipeline in your config file.
  'data pipeline in your config file.', UserWarning)
2022-09-05 10:57:16,806 - mmdeploy - WARNING - DeprecationWarning: get_onnx_config will be deprecated in the future.
2022-09-05 10:57:16,806 - mmdeploy - INFO - Export PyTorch model to ONNX: work_dir\end2end.onnx.
2022-09-05 10:57:17,523 - mmdeploy - WARNING - Can not find torch._C._jit_pass_onnx_deduplicate_initializers, function rewrite will not be applied
e:\mwdeeplearning\3.ai_software\deeplearning_deploy\deeplearning_train\mmdeploy\mmdeploy\core\optimizers\function_marker.py:158: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  ys_shape = tuple(int(s) for s in ys.shape)
e:\mwdeeplearning\3.ai_software\deeplearning_deploy\deeplearning_train\mmdetection\mmdet\models\dense_heads\anchor_head.py:123: UserWarning: DeprecationWarning: anchor_generator is deprecated, please use "prior_generator" instead
  warnings.warn('DeprecationWarning: anchor_generator is deprecated, '
e:\mwdeeplearning\3.ai_software\deeplearning_deploy\deeplearning_train\mmdetection\mmdet\core\anchor\anchor_generator.py:333: UserWarning: ``grid_anchors`` would be deprecated soon. Please use ``grid_priors``
  warnings.warn('``grid_anchors`` would be deprecated soon. '
e:\mwdeeplearning\3.ai_software\deeplearning_deploy\deeplearning_train\mmdetection\mmdet\core\anchor\anchor_generator.py:370: UserWarning: ``single_level_grid_anchors`` would be deprecated soon. Please use ``single_level_grid_priors``
  '``single_level_grid_anchors`` would be deprecated soon. '
e:\mwdeeplearning\3.ai_software\deeplearning_deploy\deeplearning_train\mmdeploy\mmdeploy\codebase\mmdet\models\dense_heads\rpn_head.py:78: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  assert cls_score.size()[-2:] == bbox_pred.size()[-2:]
e:\mwdeeplearning\3.ai_software\deeplearning_deploy\deeplearning_train\mmdeploy\mmdeploy\pytorch\functions\topk.py:57: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if k > size:
e:\mwdeeplearning\3.ai_software\deeplearning_deploy\deeplearning_train\mmdeploy\mmdeploy\codebase\mmdet\core\bbox\delta_xywh_bbox_coder.py:39: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  assert pred_bboxes.size(0) == bboxes.size(0)
e:\mwdeeplearning\3.ai_software\deeplearning_deploy\deeplearning_train\mmdeploy\mmdeploy\codebase\mmdet\core\bbox\delta_xywh_bbox_coder.py:41: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  assert pred_bboxes.size(1) == bboxes.size(1)
e:\mwdeeplearning\3.ai_software\deeplearning_deploy\deeplearning_train\mmdeploy\mmdeploy\codebase\mmdet\deploy\utils.py:93: TracerWarning: Using len to get tensor shape might cause the trace to be incorrect. Recommended usage would be tensor.shape[0]. Passing a tensor of different shape might lead to errors or silently give incorrect results.
  assert len(max_shape) == 2, '`max_shape` should be [h, w]'
e:\mwdeeplearning\3.ai_software\deeplearning_deploy\deeplearning_train\mmdeploy\mmdeploy\codebase\mmdet\core\post_processing\bbox_nms.py:259: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  dets, labels = TRTBatchedNMSop.apply(boxes, scores, int(scores.shape[-1]),
e:\mwdeeplearning\3.ai_software\deeplearning_deploy\deeplearning_train\mmdeploy\mmdeploy\mmcv\ops\nms.py:178: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  out_boxes = min(num_boxes, after_topk)
e:\mwdeeplearning\3.ai_software\deeplearning_deploy\deeplearning_train\mmdeploy\mmdeploy\mmcv\ops\nms.py:181: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  (batch_size, out_boxes)).to(scores.device))
C:\Users\Administrator\Anaconda3\envs\pytorch110cuda113\lib\site-packages\torch\onnx\symbolic_opset9.py:2819: UserWarning: Exporting aten::index operator of advanced indexing in opset 11 is achieved by combination of multiple ONNX operators, including Reshape, Transpose, Concat, and Gather. If indices include negative values, the exported graph will produce incorrect results.
  "If indices include negative values, the exported graph will produce incorrect results.")
WARNING: The shape inference of mmdeploy::TRTBatchedNMS type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of mmdeploy::TRTBatchedNMS type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of mmdeploy::MMCVMultiLevelRoiAlign type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of mmdeploy::TRTBatchedNMS type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of mmdeploy::TRTBatchedNMS type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of mmdeploy::MMCVMultiLevelRoiAlign type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of mmdeploy::TRTBatchedNMS type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of mmdeploy::TRTBatchedNMS type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of mmdeploy::MMCVMultiLevelRoiAlign type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of mmdeploy::TRTBatchedNMS type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of mmdeploy::TRTBatchedNMS type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of mmdeploy::MMCVMultiLevelRoiAlign type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of mmdeploy::TRTBatchedNMS type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of mmdeploy::TRTBatchedNMS type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of mmdeploy::MMCVMultiLevelRoiAlign type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of mmdeploy::TRTBatchedNMS type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of mmdeploy::TRTBatchedNMS type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of mmdeploy::MMCVMultiLevelRoiAlign type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
2022-09-05 10:57:30,443 - mmdeploy - INFO - Execute onnx optimize passes.
2022-09-05 10:57:30,444 - mmdeploy - WARNING - Can not optimize model, please build torchscipt extension.
More details: https://github.com/open-mmlab/mmdeploy/blob/master/docs/en/experimental/onnx_optimizer.md
2022-09-05 10:57:32,643 - mmdeploy - INFO - Finish pipeline mmdeploy.apis.pytorch2onnx.torch2onnx
loading mmdeploy_execution ...
failed to load library mmdeploy_execution
loading mmdeploy_cpu_device ...
failed to load library mmdeploy_cpu_device
loading mmdeploy_cuda_device ...
failed to load library mmdeploy_cuda_device
loading mmdeploy_graph ...
failed to load library mmdeploy_graph
loading mmdeploy_directory_model ...
failed to load library mmdeploy_directory_model
loading mmdeploy_transform ...
failed to load library mmdeploy_transform
loading mmdeploy_cpu_transform_impl ...
failed to load library mmdeploy_cpu_transform_impl
loading mmdeploy_cuda_transform_impl ...
failed to load library mmdeploy_cuda_transform_impl
loading mmdeploy_transform_module ...
failed to load library mmdeploy_transform_module
loading mmdeploy_trt_net ...
failed to load library mmdeploy_trt_net
loading mmdeploy_net_module ...
failed to load library mmdeploy_net_module
loading mmdeploy_mmcls ...
failed to load library mmdeploy_mmcls
loading mmdeploy_mmdet ...
failed to load library mmdeploy_mmdet
loading mmdeploy_mmseg ...
failed to load library mmdeploy_mmseg
loading mmdeploy_mmocr ...
failed to load library mmdeploy_mmocr
loading mmdeploy_mmedit ...
failed to load library mmdeploy_mmedit
loading mmdeploy_mmpose ...
failed to load library mmdeploy_mmpose
loading mmdeploy_mmrotate ...
failed to load library mmdeploy_mmrotate
2022-09-05 10:57:37,785 - mmdeploy - INFO - Start pipeline mmdeploy.backend.tensorrt.onnx2tensorrt.onnx2tensorrt in subprocess
2022-09-05 10:57:38,029 - mmdeploy - INFO - Successfully loaded tensorrt plugins from e:\mwdeeplearning\3.ai_software\deeplearning_deploy\deeplearning_train\mmdeploy\mmdeploy\lib\mmdeploy_tensorrt_ops.dll
[09/05/2022-10:57:38] [TRT] [I] [MemUsageChange] Init CUDA: CPU +528, GPU +0, now: CPU 11744, GPU 1281 (MiB)
[09/05/2022-10:57:38] [TRT] [I] [MemUsageSnapshot] Begin constructing builder kernel library: CPU 11816 MiB, GPU 1281 MiB
[09/05/2022-10:57:39] [TRT] [I] [MemUsageSnapshot] End constructing builder kernel library: CPU 11968 MiB, GPU 1325 MiB
[09/05/2022-10:57:39] [TRT] [W] onnx2trt_utils.cpp:366: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[09/05/2022-10:57:39] [TRT] [W] onnx2trt_utils.cpp:392: One or more weights outside the range of INT32 was clamped
[09/05/2022-10:57:43] [TRT] [W] onnx2trt_utils.cpp:392: One or more weights outside the range of INT32 was clamped
[09/05/2022-10:57:43] [TRT] [W] onnx2trt_utils.cpp:392: One or more weights outside the range of INT32 was clamped
[09/05/2022-10:57:44] [TRT] [I] No importer registered for op: TRTBatchedNMS. Attempting to import as plugin.
[09/05/2022-10:57:44] [TRT] [I] Searching for plugin: TRTBatchedNMS, plugin_version: 1, plugin_namespace:
[09/05/2022-10:57:44] [TRT] [I] Successfully created plugin: TRTBatchedNMS
[09/05/2022-10:57:44] [TRT] [I] No importer registered for op: MMCVMultiLevelRoiAlign. Attempting to import as plugin.
[09/05/2022-10:57:44] [TRT] [I] Searching for plugin: MMCVMultiLevelRoiAlign, plugin_version: 1, plugin_namespace:
[09/05/2022-10:57:44] [TRT] [I] Successfully created plugin: MMCVMultiLevelRoiAlign
[09/05/2022-10:57:44] [TRT] [W] onnx2trt_utils.cpp:392: One or more weights outside the range of INT32 was clamped
[09/05/2022-10:57:44] [TRT] [W] onnx2trt_utils.cpp:392: One or more weights outside the range of INT32 was clamped
[09/05/2022-10:57:44] [TRT] [W] onnx2trt_utils.cpp:392: One or more weights outside the range of INT32 was clamped
[09/05/2022-10:57:44] [TRT] [W] onnx2trt_utils.cpp:392: One or more weights outside the range of INT32 was clamped
[09/05/2022-10:57:44] [TRT] [W] onnx2trt_utils.cpp:392: One or more weights outside the range of INT32 was clamped
[09/05/2022-10:57:45] [TRT] [W] Tensor DataType is determined at build time for tensors not marked as input or output.
[09/05/2022-10:57:45] [TRT] [I] No importer registered for op: TRTBatchedNMS. Attempting to import as plugin.
[09/05/2022-10:57:45] [TRT] [I] Searching for plugin: TRTBatchedNMS, plugin_version: 1, plugin_namespace:
[09/05/2022-10:57:45] [TRT] [I] Successfully created plugin: TRTBatchedNMS
[09/05/2022-10:57:46] [TRT] [I] No importer registered for op: MMCVMultiLevelRoiAlign. Attempting to import as plugin.
[09/05/2022-10:57:46] [TRT] [I] Searching for plugin: MMCVMultiLevelRoiAlign, plugin_version: 1, plugin_namespace:
[09/05/2022-10:57:46] [TRT] [I] Successfully created plugin: MMCVMultiLevelRoiAlign
[09/05/2022-10:57:46] [TRT] [W] Output type must be INT32 for shape outputs
[09/05/2022-10:57:46] [TRT] [W] Output type must be INT32 for shape outputs
[09/05/2022-10:57:47] [TRT] [W] TensorRT was linked against cuBLAS/cuBLASLt 11.6.3 but loaded cuBLAS/cuBLASLt 11.4.2
[09/05/2022-10:57:47] [TRT] [I] [MemUsageChange] Init cuBLAS/cuBLASLt: CPU +728, GPU +272, now: CPU 12984, GPU 1597 (MiB)
[09/05/2022-10:57:48] [TRT] [I] [MemUsageChange] Init cuDNN: CPU +254, GPU +264, now: CPU 13238, GPU 1861 (MiB)
[09/05/2022-10:57:48] [TRT] [W] TensorRT was linked against cuDNN 8.2.1 but loaded cuDNN 8.2.0
[09/05/2022-10:57:48] [TRT] [I] Local timing cache in use. Profiling results in this builder pass will not be stored.
[09/05/2022-10:57:48] [TRT] [E] 4: [shapeCompiler.cpp::nvinfer1::builder::DynamicSlotBuilder::evaluateShapeChecks::832] Error Code 4: Internal Error (kOPT values for profile 0 violate shape constraints: reshape would change volume. IShuffleLayer Reshape_1128: reshaping failed for tensor: 1724)
Process Process-3:
Traceback (most recent call last):
  File "C:\Users\Administrator\Anaconda3\envs\pytorch110cuda113\lib\multiprocessing\process.py", line 297, in _bootstrap
    self.run()
  File "C:\Users\Administrator\Anaconda3\envs\pytorch110cuda113\lib\multiprocessing\process.py", line 99, in run
    self._target(*self._args, **self._kwargs)
  File "e:\mwdeeplearning\3.ai_software\deeplearning_deploy\deeplearning_train\mmdeploy\mmdeploy\apis\core\pipeline_manager.py", line 107, in __call__
    ret = func(*args, **kwargs)
  File "e:\mwdeeplearning\3.ai_software\deeplearning_deploy\deeplearning_train\mmdeploy\mmdeploy\backend\tensorrt\onnx2tensorrt.py", line 88, in onnx2tensorrt
    device_id=device_id)
  File "e:\mwdeeplearning\3.ai_software\deeplearning_deploy\deeplearning_train\mmdeploy\mmdeploy\backend\tensorrt\utils.py", line 153, in from_onnx
    assert engine is not None, 'Failed to create TensorRT engine'
AssertionError: Failed to create TensorRT engine
2022-09-05 10:57:49,221 - mmdeploy - ERROR - `mmdeploy.backend.tensorrt.onnx2tensorrt.onnx2tensorrt` with Call id: 1 failed. exit.

**Reproduction**

1. What command or script did you run?

```none
python ./tools/deploy.py configs/mmdet/instance-seg/instance-seg_tensorrt-fp16_dynamic-320x320-1344x1344.py   E:/MWDeepLearning/3.AI_SoftWare/DeepLearning_Deploy/DeepLearning_Train/mmdetection/configs/mask_rcnn/mask_rcnn_r50_fpn_mstrain-poly_3x_coco.py  ./checkpoints/mask_rcnn_r50_fpn_mstrain-poly_3x_coco.pth  1.jpg   --work-dir work_dir --show  --device cuda:0
```

2. Did you make any modifications on the code or config? Did you understand what you have modified?

**Environment**
python tools/check_env.py
2022-09-05 10:46:23,871 - mmdeploy - INFO -

2022-09-05 10:46:23,871 - mmdeploy - INFO - **********Environmental information**********
2022-09-05 10:46:31,039 - mmdeploy - INFO - sys.platform: win32
2022-09-05 10:46:31,039 - mmdeploy - INFO - Python: 3.7.13 (default, Mar 28 2022, 08:03:21) [MSC v.1916 64 bit (AMD64)]
2022-09-05 10:46:31,040 - mmdeploy - INFO - CUDA available: True
2022-09-05 10:46:31,041 - mmdeploy - INFO - GPU 0: NVIDIA GeForce RTX 3070
2022-09-05 10:46:31,041 - mmdeploy - INFO - CUDA_HOME: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.3
2022-09-05 10:46:31,041 - mmdeploy - INFO - NVCC: Cuda compilation tools, release 11.3, V11.3.58
2022-09-05 10:46:31,042 - mmdeploy - INFO - MSVC: 用于 x64 的 Microsoft (R) C/C++ 优化编译器 19.29.30142.1 版
2022-09-05 10:46:31,042 - mmdeploy - INFO - GCC: n/a
2022-09-05 10:46:31,042 - mmdeploy - INFO - PyTorch: 1.10.0+cu113
2022-09-05 10:46:31,042 - mmdeploy - INFO - PyTorch compiling details: PyTorch built with:
  - C++ Version: 199711
  - MSVC 192829337
  - Intel(R) Math Kernel Library Version 2020.0.2 Product Build 20200624 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.2.3 (Git Hash 7336ca9f055cf1bfa13efb658fe15dc9b41f0740)
  - OpenMP 2019
  - LAPACK is enabled (usually provided by MKL)
  - CPU capability usage: AVX512
  - CUDA Runtime 11.3
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.2
  - Magma 2.5.4
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.3, CUDNN_VERSION=8.2.0, CXX_COMPILER=C:/w/b/windows/tmp_bin/sccache-cl.exe, CXX_FLAGS=/DWIN32 /D_WINDOWS /GR /EHsc /w /bigobj -DUSE_PTHREADPOOL -openmp:experimental -IC:/w/b/windows/mkl/include -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DUSE_FBGEMM -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.10.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=OFF, USE_OPENMP=ON,

2022-09-05 10:46:31,043 - mmdeploy - INFO - TorchVision: 0.11.0+cu113
2022-09-05 10:46:31,043 - mmdeploy - INFO - OpenCV: 4.6.0
2022-09-05 10:46:31,043 - mmdeploy - INFO - MMCV: 1.5.0
2022-09-05 10:46:31,044 - mmdeploy - INFO - MMCV Compiler: MSVC 192930140
2022-09-05 10:46:31,044 - mmdeploy - INFO - MMCV CUDA Compiler: 11.3
2022-09-05 10:46:31,044 - mmdeploy - INFO - MMDeploy: 0.7.0+9918d29
2022-09-05 10:46:31,045 - mmdeploy - INFO -

2022-09-05 10:46:31,045 - mmdeploy - INFO - **********Backend information**********
2022-09-05 10:46:31,715 - mmdeploy - INFO - onnxruntime: 1.12.1 ops_is_avaliable : False
2022-09-05 10:46:31,757 - mmdeploy - INFO - tensorrt: 8.2.5.1   ops_is_avaliable : True
2022-09-05 10:46:31,804 - mmdeploy - INFO - ncnn: None  ops_is_avaliable : False
2022-09-05 10:46:31,813 - mmdeploy - INFO - pplnn_is_avaliable: False
2022-09-05 10:46:31,820 - mmdeploy - INFO - openvino_is_avaliable: False
2022-09-05 10:46:31,874 - mmdeploy - INFO - snpe_is_available: False
2022-09-05 10:46:31,874 - mmdeploy - INFO -

2022-09-05 10:46:31,875 - mmdeploy - INFO - **********Codebase information**********
2022-09-05 10:46:35,268 - mmdeploy - INFO - mmdet:      2.25.0
2022-09-05 10:46:35,268 - mmdeploy - INFO - mmseg:      0.26.0
2022-09-05 10:46:35,268 - mmdeploy - INFO - mmcls:      0.23.1
2022-09-05 10:46:35,270 - mmdeploy - INFO - mmocr:      0.6.0
2022-09-05 10:46:35,270 - mmdeploy - INFO - mmedit:     None
2022-09-05 10:46:35,271 - mmdeploy - INFO - mmdet3d:    1.0.0rc3
2022-09-05 10:46:35,271 - mmdeploy - INFO - mmpose:     None
2022-09-05 10:46:35,271 - mmdeploy - INFO - mmrotate:   None


**Error traceback**

If applicable, paste the error trackback here.

```none
A placeholder for trackback.
```

**Bug fix**

If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

--- a/mmdeploy/codebase/mmdet/models/roi_heads/test_mixins.py
+++ b/mmdeploy/codebase/mmdet/models/roi_heads/test_mixins.py
@@ -44,7 +44,6 @@ def bbox_test_mixin__simple_test_bboxes(ctx,
             rois.size(0), rois.size(1), 1)
     rois = torch.cat([batch_index, rois[..., :4]], dim=-1)
     batch_size = rois.shape[0]
-    num_proposals_per_img = rois.shape[1]
 
     # Eliminate the batch dimension
     rois = rois.view(-1, 5)
@@ -53,12 +52,10 @@ def bbox_test_mixin__simple_test_bboxes(ctx,
     bbox_pred = bbox_results['bbox_pred']
 
     # Recover the batch dimension
-    rois = rois.reshape(batch_size, num_proposals_per_img, rois.size(-1))
-    cls_score = cls_score.reshape(batch_size, num_proposals_per_img,
-                                  cls_score.size(-1))
+    rois = rois.reshape(batch_size, -1, rois.size(-1))
+    cls_score = cls_score.reshape(batch_size, -1, cls_score.size(-1))
 
-    bbox_pred = bbox_pred.reshape(batch_size, num_proposals_per_img,
-                                  bbox_pred.size(-1))
+    bbox_pred = bbox_pred.reshape(batch_size, -1, bbox_pred.size(-1))
     det_bboxes, det_labels = self.bbox_head.get_bboxes(
         rois,
         cls_score,

